src/common/grpc/src/channel_manager.rs

```diff
@@ -224,9 +224,10 @@ impl ChannelManager {
             return;
         }
 
-        let pool = self.pool.clone();
-        let _handle = common_runtime::spawn_global(async {
-            recycle_channel_in_loop(pool, RECYCLE_CHANNEL_INTERVAL_SECS).await;
+        let id = self.id;
+        let pool = Arc::downgrade(&self.pool);
+        let _handle = common_runtime::spawn_global(async move {
+            recycle_channel_in_loop(id, pool, RECYCLE_CHANNEL_INTERVAL_SECS).await;
```

Comment:
Is it expected that we have many channel managers?

non-blocking: Should we add some metrics about the active channel/channel manager? We can do it in a new PR.